Modify request & response structure (#103)
shopware/admin-api-reference: docs/concepts/endpoint-structure/request-response-structure.md

## Request format

> The request format for Shopware 6 store API's must follow the below format:

Reviewer: ```suggestion
The request format for the Shopware 6 Admin API must follow the below format:
```

* The request body has to be JSON encoded.

* It is mandatory to use type-safe values, e.g., if the API expects an integer value, you are required to provide an actual integer.

* When using a Date field, make sure to use an ISO 8601 compatible date format.

### Request URL

The Admin API can be separated into two general sets of endpoints

* Generic entity endpoints

* Specific interaction/configuration endpoints

Both fulfill specific cases which are outlined below.

#### Generic entity endpoints

> Provide CRUD functionalities on **all entities available** in Shopware. Go to the [Entity Reference](../resources/entity-reference.md) section to see a list of all available entities and their structure.

Reviewer: ```suggestion
Provide CRUD functionalities on **all entities available** in Shopware. Go to the [Entity Reference](../../resources/entity-reference.md) section to see a list of all available entities and their structure.
```

> Furthermore, these endpoints can be divided into **read** and **write** operations. To learn more about those, please head to the corresponding guide on [Reading Entities](../guides/reading-entities.md) and [Writing Entities](../guides/writing-entities/README.md). These endpoints are generic, since they are entirely based on the entities definitions and contain no additional business logic besides data validation.

Reviewer: ```suggestion
Furthermore, these endpoints can be divided into **read** and **write** operations. To learn more about those, please head to the corresponding guide on [Reading Entities](reading-entities.md) and [Writing Entities](writing-entities/README.md). These endpoints are generic, since they are entirely based on the entity definitions and contain no additional business logic besides data validation.
```

> The URL structure for *Generic entity endpoints* is:
> ```
> /api/{entity-name}(/{entity-id})
> ```

#### Specific interaction or configuration endpoints

Provide interactions for more sophisticated operations which can change the state of the system but are not directly based on entities alone. We differ between more than 20 categories of about 140 specific endpoints, such as

* Document Management
* Order State Management
* User Role Management
* System Operations
* Authorization & Authentication

which will let you automate every part of the store operation.

> The URL structure for *Specific interaction or configuration endpoints* roughly follows this schema:
> ```
> /api/_action/{interaction-name}
> ```

### Request headers

> Request headers provide information about your request to a REST API service that allows you to authenticate/authorize and receive the request body.

Reviewer: I would remove this from the ADMIN api since the authorization does not happen via headers, but the auth endpoint with a USER or Integration and from there with a bearer token

| Request header              | Key                                    | Description                                                                                |
| --------------------------- | ---------------------------------------|--------------------------------------------------------------------------------------------|

> | sw-access-token             | SWSCXYZSNTVMAWNZDNFKSHLAYW             | Access key to identify and get authorized to respective sales channel access               |

Reviewer: ```suggestion

```

Reviewer: The admin API does not use the access or context token

> | sw-context-token            | 84266fdbd31d4c2c6d0665f7e8380fa3       | Endpoints that require user authentication                                                 |

Reviewer: ```suggestion

```

> | Content-Type                | application/json                       | Indicate the format of the request body                                                    |

Reviewer: ```suggestion
| Content-Type                | application/json or application/vnd.api+json                       | Indicate the format of the request body                                                    |
```
